packages/core/src/sync-realtime/service.ts

```diff
@@ -8,878 +8,633 @@ import {
-   */
-  reconcileReorg = async () => {
-    const hasDeepReorg = await this.validateFinalizedBlock();
-    if (hasDeepReorg) return;
-
-    // Find common ancestor block by requesting remote block until you find one that fits in the local chain.
-    while (this.blocks.length > 0) {
-      const localBlock = this.blocks.pop();
-      const remoteBlock = await this._eth_getBlockByNumber(localBlock!.number);
-      const parent =
-        this.blocks.length === 0
-          ? this.finalizedBlock
-          : this.blocks[this.blocks.length - 1];
-
-      if (parent.hash === remoteBlock.parentHash) {
-        this.common.logger.trace({
-          service: "realtime",
-          msg: `Found common ancestor block ${parent.number} (network=${this.network.name})`,
-        });
-
-        this.logs = this.logs.filter((log) => log.blockNumber <= parent.number);
-
-        await this.syncStore.deleteRealtimeData({
-          chainId: this.network.chainId,
-          fromBlock: BigInt(parent.number),
-        });
-
-        this.emit("reorg", {
-          ...maxCheckpoint,
-          blockTimestamp: parent.timestamp,
-          chainId: this.network.chainId,
-          blockNumber: parent.number,
-        });
-
-        this.common.logger.warn({
-          service: "realtime",
-          msg: `Detected reorg with common ancestor ${parent.number} (network=${this.network.name})`,
-        });
-
-        return;
-      }
-    }
-
-    throw new Error(
-      "Invariant violated: Unable to find common ancestor block in local chain.",
-    );
-  };
-
-  /**
-   * Check if deep re-org occured by comparing remote "finalized" block to local.
-   */
-  private validateFinalizedBlock = async () => {
-    const remoteFinalizedBlock = await this._eth_getBlockByNumber(
-      this.finalizedBlock.number,
+    realtimeSyncService.localChain = realtimeSyncService.localChain.filter(
+      ({ block }) => block.number > pendingFinalizedBlock.number,
     );
 
-    if (remoteFinalizedBlock.hash !== this.finalizedBlock.hash) {
-      const msg = `Detected unrecoverable reorg at block ${this.finalizedBlock.number} with local hash ${this.finalizedBlock.hash} and remote hash ${remoteFinalizedBlock.hash} (network=${this.network.name})`;
-      this.common.logger.warn({ service: "realtime", msg });
-
-      this.emit("fatal", new Error(msg));
-
-      this.blocks = [];
-      this.logs = [];
-
-      this.finalizedBlock = realtimeBlockToLightBlock(remoteFinalizedBlock);
-
-      return true;
-    }
-
-    this.common.logger.trace({
-      service: "realtime",
-      msg: `Confirmed local hash matches remote hash at finalized block number ${this.finalizedBlock.number} (network=${this.network.name})`,
+    // TODO: Update this to insert:
+    // 1) Log filter intervals
+    // 2) Factory contract intervals
+    // 3) Child filter intervals
+    await realtimeSyncService.syncStore.insertRealtimeInterval({
```

Comment:
Worth noting that you mutate the `realtimeSyncService.localChain` ABOVE this database query, which seems dangerous.